Lib/gzip.py

```diff
@@ -587,7 +587,7 @@ def compress(data, compresslevel=_COMPRESS_LEVEL_BEST, *, mtime=None):
     header = _create_simple_gzip_header(compresslevel, mtime)
     trailer = struct.pack("<LL", zlib.crc32(data), (len(data) & 0xffffffff))
     # Wbits=-15 creates a raw deflate block.
-    return header + zlib.compress(data, wbits=-15) + trailer
+    return header + zlib.compress(data, compresslevel, wbits=-15) + trailer
```

Comment:
According to the docs, `level` arg is keyword-only.

Reply:
Hmm, the docs seem to be wrong though. It works as positional. However on line 586 it's used as kw arg, so it's a bit better to do that for consistency.

Reply:
Since you thoroughly reviewed whether this one word is correctly added to the code, may I kindly request that you push the merge button? Thank you for your review!

Reply:
Can you look into adding a unit test for this?

I can't merge the PR (not a core dev), but I can label it for core dev review, if you can add a unit test.

Reply:
No I can't. A raw DEFLATE stream does not contain any information on its compression level. And that level 9 is smaller than level 1 is not a given depending on the test data. So there is not a way to reliable ascertain that the correct compression level is used.

What I can test is whether the XFL is set correctly, but that is already in the test suite. That is a field in the gzip header though, so it is not affected by this change.

Reply:
I believe that what you can do here is strip header (10 chars for data i tested), and compare the remaining data between two runs with differing compression levels. If that returned data is different, the test should pass.

Reply:
You got me thinking. I made a false statement here:

> No I can't. A raw DEFLATE stream does not contain any information on its compression level. 

Because raw deflate does contain a `BTYPE` that notes what kind of stream it is. And there is a special btype for an uncompressed stream (compression level 0), so it could indeed be tested. I will add it right away.

Reply:
A regression test was added. This takes advantage of the fact that a stream compressed with compresslevel 0 , should still contain the original data.